tools/tracing/timeline/README.md

```diff
@@ -0,0 +1,124 @@
+# MMTk GC visualization
+
+This directory contains tools for visualizing the execution time of each work packet on a timeline. 
+
+## Before Running
+
+Before running, you should make sure the [bpftrace] command line utility is installed.
+
+[bpftrace]: https://github.com/iovisor/bpftrace
+
+## Capture a log
+
+Run the `./capture.py` script to capture a log.
+
+In this example, we use the OpenJDK binding to run the `lusearch` benchmark in the DaCapo Benchmark
+Suite.
+
+Run the following command with a **normal** user (*not* as `root` or using `sudo`):
+
+```shell
+./capture.py -e 50 -m /path/to/libmmtk_openjdk.so
+```
+
+`-e 50` means we only capture one GC in every 50 GCs because otherwise it will record too much log.
```

Comment:
I think we should clarify this.

The main problem is that the printing in bpftrace is done via a fixed-size user/kernel space buffer (similar to ftrace's `trace_printk` and `trace_pipe`), and too much logging will overflow the buffer and cause events to be dropped. This is explained at the end of page 88 of the paper.

The `-e` workaround helps when there are too frequent back to back GCs, but still assumes that we won't overflow that buffer during a single GC (which more or less works for most workloads that we tested). If even very large `-e` is insufficient, users might want to consider setting `BPFTRACE_PERF_RB_PAGES`.

Reply:
I added some clarification in that paragraph.